components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigServiceImpl.java

```diff
@@ -77,6 +77,13 @@ public void addUserStore(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtE
 
         loadTenant();
         try {
+            List<String> reservedUserStores = IdentityUtil.getPropertyAsList(UserStoreConfigurationConstant
```

Comment:
Don't we have the backend support to update the userstore name?
If update support is there, need to do the same check under update operation as well?

Reply:
I've tried to update the domain name via the REST API. It's returning an invalid input error, seems updating domain name is restricted.